### PR TITLE
IGNITE-14120 fix multiple results bug when query parallelism is enabled.

### PR DESCRIPTION
**Pull request by vladErmakov07**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by vladErmakov07** [comment]
@korlov42 could you please take a look?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+/**
+ * IGNITE-14120 Test for correct results in case of query with single partition and cache with parallelism > 1
+ */
+public class IgniteSqlSinglePartitionMultiParallelismTest extends AbstractIndexingCommonTest {
+    /** */
+    private static final String CACHE_NAME = "SC_NULL_TEST";
+
+    /** Test timeout. */
+    @Rule
+    public Timeout globalTimeout = new Timeout((int)GridTestUtils.DFLT_TEST_TIMEOUT);
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:43]: test timeout is already handled by GridAbstractTest#runRule

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        startGrids(1);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        grid(0).destroyCaches(Collections.singletonList(CACHE_NAME));
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:54]: is it really necessary to recreate the table for every test?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.cache.CacheMode.PARTITIONED;
+import static org.apache.ignite.cache.PartitionLossPolicy.READ_WRITE_SAFE;
+
+/**
+ * IGNITE-14120 Test for correct results in case of query with single partition and cache with parallelism > 1
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:35]: missed dot at the end

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java (hunk 1 above)
Comment (vladErmakov07) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java]: Thx, removed it.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java (hunk 2 above)
Comment (vladErmakov07) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:47]: Looked at this closely, it is not necessary. Moved it to beforeTestsStarted function

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java (hunk 3 above)
Comment (vladErmakov07) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java]: Fixed

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+            Long res = (Long) results.get(0).get(0);
+
+            assertEquals(1, results.size());
+            assertEquals(Long.valueOf(2), res);
+        });
+    }
+
+    /** */
+    private void run(Runnable test) throws Exception {
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:127]: this method is useless now

Comment (vladErmakov07) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java]: Forgot about it, removed.

**Pull request merged**